### PR TITLE
Implement contract history and completion tracker

### PR DESCRIPTION
**Pull request by moonysolari**
- A tick will now appear on the top-left of the contract tile if the contract has been completed before by the user. 
- A cross will appear instead if the contract has been attempted before by the player but never completed (i.e. failed).
- The "My History" menu now correctly displays the player's history of played contracts, sorted from most to least recent. 
- The "My History", "My Contracts", and "My Favorites" menus now support pressing E to only show completed, failed, or all contracts. All three views work. Previously configured views will persist between different user visits to the page.
- Tested to be working on all three game versions.

### CODE REVIEW

File: static/UserDefault.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "LinkedAccounts": {},
     "Extensions": {
         "PeacockEscalations": {},
-        "PeacockFavoriteContracts": [],
```
Comment (RDIL) [inline static/UserDefault.json:6]: Favorite contracts have their own system, this will break it.

**Review by RDIL — CHANGES_REQUESTED** [review]

File: static/UserDefault.json (hunk 1 above)
Comment (moonysolari) [reply static/UserDefault.json:6]: Could you please elaborate?

Comment (moonysolari) [reply static/UserDefault.json:6]: I added it back. Deleting it was an oversight.

**Review by RDIL — APPROVED** [review]

**Comment by RDIL** [comment]
On standby until after v6 stable.

**Pull request merged**